### PR TITLE
8300589: Use @snippet and @linkplain in java.text.CollationKey and java.text.CompactNumberFormat

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
This PR implements _JEP 413: Code Snippets in Java API Documentation_ for [java.text.CollationKey](https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/text/CollationKey.html) and [java.text.CompactNumberFormat](https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/text/CompactNumberFormat.html).

Code examples using \<pre> ... \</pre> blocks are replaced with the @ snippet syntax where applicable.

Additionally, for [java.text.CompactNumberFormat](https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/text/CompactNumberFormat.html) internal plain text links using \<a> ... \</a> are replaced with @ linkplain. (External links are kept the same)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300589](https://bugs.openjdk.org/browse/JDK-8300589): Use @snippet and @linkplain in java.text.CollationKey and java.text.CompactNumberFormat

### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12108/head:pull/12108` \
`$ git checkout pull/12108`

Update a local copy of the PR: \
`$ git checkout pull/12108` \
`$ git pull https://git.openjdk.org/jdk pull/12108/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12108`

View PR using the GUI difftool: \
`$ git pr show -t 12108`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12108.diff">https://git.openjdk.org/jdk/pull/12108.diff</a>

</details>

### CODE REVIEW

**Review by LanceAndersen — APPROVED** [review]

File: src/java.base/share/classes/java/text/CompactNumberFormat.java [hunk 1]
```diff
@@ -384,8 +384,8 @@ public final class CompactNumberFormat extends NumberFormat {
      * @param decimalPattern a decimal pattern for general number formatting
      * @param symbols the set of symbols to be used
      * @param compactPatterns an array of
-     *        <a href = "CompactNumberFormat.html#compact_number_patterns">
-     *        compact number patterns</a>
+     *        {@linkplain CompactNumberFormat##compact_number_patterns
```
Comment (naotoj) [inline src/java.base/share/classes/java/text/CompactNumberFormat.java:387]: Can eliminate `CompactNumberFormat` here.

File: src/java.base/share/classes/java/text/CompactNumberFormat.java [hunk 2]
```diff
@@ -412,8 +412,8 @@ public CompactNumberFormat(String decimalPattern,
...
```
Comment (naotoj) [inline src/java.base/share/classes/java/text/CompactNumberFormat.java:415]: Same as above.

**Review by naotoj — COMMENTED** [review]
Looks good. Minor suggestions follow.

File: src/java.base/share/classes/java/text/CompactNumberFormat.java (hunk 1 above)
Comment (justin-curtis-lu) [reply src/java.base/share/classes/java/text/CompactNumberFormat.java:387]: Ah good point, it's the same file. Will make the change, thanks!

**Review by naotoj — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]

**Review by irisclark — APPROVED** [review]

**Pull request closed without merging**